subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AbstractGradleProfilerBuildExperimentRunner.java

```diff
@@ -53,12 +54,22 @@ public AbstractGradleProfilerBuildExperimentRunner(BenchmarkResultCollector resu
         this.flameGraphGenerator = jfrProfileTargetDir == null
             ? ProfilerFlameGraphGenerator.NOOP
             : new JfrFlameGraphGenerator(new File(jfrProfileTargetDir));
-        this.profiler = jfrProfileTargetDir == null
-            ? org.gradle.profiler.Profiler.NONE
-            : ProfilerFactory.of(Arrays.asList("jfr")).createFromOptions(new OptionParser().parse());
+        this.profiler = createProfiler(jfrProfileTargetDir);
         this.resultCollector = resultCollector;
     }
 
+    private org.gradle.profiler.Profiler createProfiler(String jfrProfileTargetDir) {
```

Comment:
I already merged this change to `master`, you can rebase it.

Reply:
Rebased.